sphinx/ext/javadoctest.py

```diff
@@ -0,0 +1,118 @@
+"""Mimic javadoctest in Sphinx.
+
+The extension automatically execute code snippets and checks their results on Java engine.
+"""
+
+import subprocess
+from typing import Any, Dict
+
+from sphinx.application import Sphinx
+from sphinx.ext.doctest import (
+    DocTestBuilder,
+    TestcodeDirective,
+    TestoutputDirective,
+    doctest,
+    sphinx,
+)
+from sphinx.locale import __
+
+
+class JavaTestcodeDirective(TestcodeDirective):
```

Comment:
Maybe `JavaTestcodeDirective` -> ``JavaTestCodeDirective`? 

In addition, the directive could support different versions of Java instead of only supporting Java 11. This could be configured via `:version: 11` for instance.

Reply:
> Maybe `JavaTestcodeDirective` -> `JavaTestCodeDirective`?
> 
> In addition, the directive could support different versions of Java instead of only supporting Java 11. This could be configured via `:version: 11` for instance.

Currently it is tested with ny Java version >= 11. it depende about what version was configured intn you JAVA_HOME.

Reply:
I agree with that, but since Java 11 is not yet deprecated, people might write code that is *not* compatible with some Java version (e.g., Java 12 adds more methods to the `String` interface that are not present in Java 11). That way, if the current Java version is not installed on the OS running the tested code, then the snippet is not executed at all and there are not errors (I would suggest emitting a warning instead).

Reply:
Just added logic to validate Java/Maven needed by flavors. There is also pending to add a new feature thru `doctest` to handle `javaversion` as the same that [pyversion](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html#:~:text=pyversion%2C%20a%20string%20option%2C%20can%20be%20used%20to%20specify%20the%20required%20Python%20version%20for%20the%20example%20to%20be%20tested.%20For%20instance%2C%20in%20the%20following%20case%20the%20example%20will%20be%20tested%20only%20for%20Python%20versions%20greater%20than%203.3%3A) is working, but this will be consider as new feature in another PR